modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/TypedInput.es.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import ClaySelect from '../shared/ClaySelect.es';
+
+class TypedInput extends React.Component {
+	constructor(props) {
+		super(props);
+		this._onChange = this._onChange.bind(this);
+	}
+	_onChange(event) {
+		const {onChange} = this.props;
+		const value = event.target.value;
+
+		if (onChange) {
+			onChange(value);
+		}
+	}
+	render() {
+		const {value, options, className} = this.props;
+		const isOption = options.length;
+
+		return (
+			<div className={className}>
+				{
+					(isOption === 0) ?
```

Comment:
Source formatter might complain about a variable starting with `is`, but if we're naming it `isOption` I'd expect this to be a boolean. We can do `!!options.length` to convert the value into a boolean and this check would read more human-like too with `isOption ? `

If you want to keep the variable name, we could instead make this a function and call `this.isOption()`.